### ISSUE TITLE
vault read format

### ISSUE THREAD

**Issue by possani** [hashicorp/vault#4709]
Please post questions on the mailing list (https://groups.google.com/forum/m/#!forum/vault-tool) rather than in GitHub issues.

With over 1700 subscribers, the mailing list gets a _lot_ more eyes on any particular question. You'll get an answer faster, not slower, by taking advantage of the suggestions and answers provided by the wealth of experience that long-time users and devs bring to the list.

It also prevents problematic issue workflows for questions where either the issue is closed immediately but then discussion continues on a closed issue; or, the issue is not closed, discussion dies down, and the issue remains open indefinitely waiting for further potential replies.

Thanks!

**Comment by possani** [comment]
Hi,

I started using Vault some days ago. I am trying to read the id of a role that I just created and format it as json, but I get the following error:

vault read auth/approle/role/<ROLE>/role-id -format=json
Too many arguments (expected 1, got 2)

If I try to extract a field I get the same error.

According to the information available here:
https://www.vaultproject.io/docs/commands/read.html

I should be able to use those flags.

Am I doing something wrong?
